integration/mediation-tests/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/servers/GreenMailServer.java

```diff
@@ -0,0 +1,237 @@
+            }
+        }
+        return emailReceived;
+    }
+
+    /**
+     * Check inbox and make sure a particular email is deleted
+     *
+     * @param emailSubject
+     * @param protocol
+     * @return
+     * @throws ESBMailTransportIntegrationTestException
+     */
+    public static boolean checkEmailDeleted(String emailSubject, String protocol)
+            throws ESBMailTransportIntegrationTestException, MessagingException {
+        boolean isEmailDeleted = false;
+        long startTime = System.currentTimeMillis();
+
+        while ((System.currentTimeMillis() - startTime) < WAIT_TIME_MS) {
+            if (!isMailReceivedBySubject(emailSubject, EMAIL_INBOX, protocol)) {
+                log.info("Email has been deleted successfully!");
+                isEmailDeleted = true;
+                break;
+            }
+            try {
+                Thread.sleep(500);
+            } catch (InterruptedException e) {
+                log.warn("Error during thread sleep for 0.5 sec");
+            }
+        }
+        return isEmailDeleted;
+    }
+
+    /**
+     * Check trash and make sure a particular email is moved to trash
+     *
+     * @param emailSubject
+     * @param protocol
+     * @return
+     * @throws ESBMailTransportIntegrationTestException
+     */
+    public static boolean checkEmailMoved(String emailSubject, String protocol)
+            throws ESBMailTransportIntegrationTestException, MessagingException {
+        boolean mailReceived = false;
+        long startTime = System.currentTimeMillis();
+        while ((System.currentTimeMillis() - startTime) < WAIT_TIME_MS) {
+            if (isMailReceivedBySubject(emailSubject, EMAIL_TRASH, protocol)) {
+                log.info("Found the moved email in mailbox : " + emailSubject);
+                mailReceived = true;
+                break;
+            }
+            try {
+                Thread.sleep(500);
+            } catch (InterruptedException e) {
+                log.warn("Error during thread sleep for 0.5 sec");
+            }
+        }
+        return mailReceived;
+    }
+
+    /**
+     * Delete all unread emails in the inbox
+     *
+     * @param protocol
+     * @throws ESBMailTransportIntegrationTestException
+     */
+    public static void deleteAllEmails(String protocol)
+            throws ESBMailTransportIntegrationTestException, MessagingException {
+        Folder inbox = null;
+        Store store;
+        if ("pop3".equals(protocol)) {
+            store = getPopConnection();
+        } else {
+            store = getImapConnection();
+        }
+        inbox = store.getFolder(EMAIL_INBOX);
+        inbox.open(Folder.READ_WRITE);
+        Message[] messages = inbox.getMessages();
+
+        for (Message message : messages) {
+            message.setFlag(Flags.Flag.DELETED, true);
+            log.info("Deleted email Subject : " + message.getSubject());
+        }
+        if (inbox != null) {
+            inbox.close(true);
```

Comment:
Shouldn't this done in a finally block? what will happen if an exception occurred before that?

Reply:
fixed